app/jobs/course/material/download_job.rb

```diff
@@ -0,0 +1,31 @@
+class Course::Material::DownloadJob < ApplicationJob
+  include TrackableJob
+
+  protected
+
+  # Performs the download service.
+  #
+  # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+  #   downloaded.
+  # @param [String] filename The name of the zip file, it should be set to the name of the folder
+  #   containing the contents.
+  def perform_tracked(contents, filename)
+    zip_file = Course::Material::DownloadService.download_and_zip(contents)
+    redirect_to publish_file(zip_file, filename.gsub(/[:\/\\]/, ' ') + '.zip')
```

Comment:
abstract out the filename processing for safety to a function, because that's what I think you're doing